clippy_lints/src/matches/manual_unwrap_or.rs

```diff
@@ -48,7 +48,7 @@ fn get_none<'tcx>(cx: &LateContext<'_>, arm: &Arm<'tcx>) -> Option<&'tcx Expr<'t
         && cx.tcx.lang_items().get(LangItem::ResultErr) == Some(def_id)
     {
         Some(arm.body)
-    } else if let PatKind::Wild = arm.pat.kind {
+    } else if let (PatKind::Wild, true) = (arm.pat.kind, allow_wildcard) {
```

Comment:
This statement isn't as clear as it should be for a simple if check (doing pattern destructuring for a simple boolean is a bit weird)


```suggestion
    } else if let PatKind::Wild = arm.pat.kind
        && allow_wildcard
        {
```